applications/portal/backend/api/views/rest/experiment.py

```diff
@@ -217,3 +224,73 @@ def download_populations(self, request, pk=None, active_populations=None):
 
     def perform_create(self, serializer):
         experiment = serializer.save(owner=self.request.user)
+
+    def check_if_user_is_owner(self, kwargs_experiment_id):
+        try:
+            user = get_current_user()
+            experiment = Experiment.objects.get(id=kwargs_experiment_id)
+            if user.id != experiment.owner.id:
+                return Response(status=status.HTTP_403_FORBIDDEN, data={'detail': 'You are not the owner of the experiment'})
+        except Experiment.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': 'Experiment does not exist'})
+        
+    def check_if_subpopulation_exists(self, subpopulation_id, kwargs_experiment_id):
+        try:
+            sub_population_object = Population.objects.get(id=subpopulation_id)
+            is_residential = sub_population_object.experiment is None
+            if is_residential:
+                return Response(status=status.HTTP_403_FORBIDDEN, data={'detail': 'You cannot change the name of a residential population'})
+            if int(sub_population_object.experiment.id) != int(kwargs_experiment_id):
+                return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': f'Subpopulation with id {subpopulation_id} does not exist in this experiment'})
+        except Population.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND, data={'detail': f'Subpopulation with id {subpopulation_id} does not exist'})
+        return sub_population_object
+    
+
+    @action(
+        detail=True, 
+        methods=['patch'], 
+        parser_classes=(JSONParser,),
+        url_name="rename_population"
+    )
+    def rename_population(self, request, **kwargs):
+        kwargs_experiment_id = kwargs['pk']
+        is_user_owner = self.check_if_user_is_owner(kwargs_experiment_id)
+        request_data = json.loads(request.body)
+        sub_populations = []
+        try:
+            for change in request_data['change']:
+                subpopulation_id = change['pid']
+                get_subpopulation_if_valid = self.check_if_subpopulation_exists(subpopulation_id, kwargs_experiment_id)
+                get_subpopulation_if_valid.name = change['new_name']
+                sub_populations.append(get_subpopulation_if_valid)
+            
+            for sub_population_object in sub_populations:
+                sub_population_object.save()
+        except KeyError:
+            return Response(status=status.HTTP_400_BAD_REQUEST, data={'detail': 'Invalid request body'})
+
+        return Response(status=status.HTTP_200_OK, data={'detail': 'Population name changed successfully'})
+
+    @action(
+        detail=True, methods=["patch"], 
+        parser_classes=(JSONParser,),
+        url_name="rename_subpopulation"
+    )
+    def rename_subpopulation(self, request, **kwargs):
+        kwargs_experiment_id = kwargs['pk']
```

Comment:
@D-GopalKrishna isn't this the same a patch a population?

Reply:
It's a different API call. 

In the following population in Frontend.
> ABC@DEF
> ABC@XYZ

We can change both ABC, and DEF. When DEF is changed, we make call to `rename_subpopulation`, and when ABC is changed, we call `rename_population`, and it sends a list of all the subpopulation to rename.

Reply:
imo this is logic for the backend, depending the name change it knows (if there is a @) if it's a population or a "sub"population. Remember sub populations are plain populations